### PR TITLE
Fix disposal units flushing too soon after power-on

### PR DESCRIPTION
**Pull request by eoineoineoin**
## About the PR

Fix disposal units flushing immediately on power-on.

## Why / Balance

When a disposal unit was powered on while engaged, the unit would flush immediately, skipping the actual wait time for the flush to occur. Combined with having a power-toggle button on the UI, ProGamers(TM) could use this to flush items real fast:

https://github.com/user-attachments/assets/4602bbcb-8aea-4355-b834-91a6688cd363

Pretty clear this is unintended behaviour. Now, when the unit is powered on while engaged, will time the flush from the point of power-on. 

## Requirements

- [X] I have read and am following the [Pull Request and Changelog Guidelines](https://docs.spacestation14.com/en/general-development/codebase-info/pull-request-guidelines.html).
- [X] I have added media to this PR or it does not require an ingame showcase.

## Breaking changes

**Changelog**

### CODE REVIEW

**Comment by slarticodefast** [comment]
Would make more sense imo if there was no pressure buildup without power and you could flush anytime if pressure was full. The wait time always feels so random to me.

**Comment by lzk228** [comment]
feature!

**Comment by Emisse** [comment]
disposalers in shambles

**Comment by Ilya246** [comment]
maybe this but also just make them flush immediately on flush button press

**Comment by Ilya246** [comment]
> > maybe this but also just make them flush immediately on flush button press
> 
> Isn't that a feature of emagged disposal bins

no they just recharge pressure instantly

File: Content.Server/Disposal/Unit/EntitySystems/DisposalUnitSystem.cs [hunk 1]
```diff
@@ -320,9 +320,10 @@ private void OnPowerChange(EntityUid uid, SharedDisposalUnitComponent component,
             return;
         }
 
-        if (component.Engaged && !TryFlush(uid, component))
+        if (component.Engaged)
         {
-            QueueAutomaticEngage(uid, component);
+            // Run ManualEngage to reclaculate a new flush time
+            ManualEngage(uid, component);
```
Comment (metalgearsloth) [inline Content.Server/Disposal/Unit/EntitySystems/DisposalUnitSystem.cs:326]:
```suggestion
            // Run ManualEngage to recalculate a new flush time
            ManualEngage(uid, component);
```

**Pull request merged**